fuzz_lightyear/datastore.py

```diff
@@ -185,7 +189,11 @@ def wrapped(*args: Any, **kwargs: Any) -> Any:
             if arg_name not in mapping:
                 raise TypeError
 
-            value = mapping[arg_name]()
+            # We don't allow specific endpoints here, and only use the default.
+            # If we don't have a default value, then we raise an error.
+            value = mapping[arg_name].default_factory()()
+            if value is None:
+                raise TypeError
```

Comment:
You might need to perform the check against `_none_func` here, since this won't support the use case of a factory explicitly returning `None`.